### PR TITLE
Update settings UI

### PR DESCRIPTION
**Pull request by zerts**

### CODE REVIEW

**Comment by zerts** [comment]
Here is updated styles for Toggle component: https://github.com/zeriontech/components/pull/1

File: src/ui/ui-kit/Frame/index.ts
Comment (everdimension) [inline src/ui/ui-kit/Frame/index.ts:1]: Empty file, export { Frame } here

File: src/ui/pages/Security/AutoLockTimer.tsx [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
+import React, { useState } from 'react';
 import type { GlobalPreferences } from 'src/shared/types/GlobalPreferences';
+import { Background } from 'src/ui/components/Background';
 import { NavigationTitle } from 'src/ui/components/NavigationTitle';
 import { PageBottom } from 'src/ui/components/PageBottom';
 import { PageColumn } from 'src/ui/components/PageColumn';
 import { PageTop } from 'src/ui/components/PageTop';
 import { useGlobalPreferences } from 'src/ui/features/preferences/usePreferences';
+import { Button } from 'src/ui/ui-kit/Button';
+import { Frame } from 'src/ui/ui-kit/Frame/Frame';
```
Comment (everdimension) [inline src/ui/pages/Security/AutoLockTimer.tsx:11]:
```suggestion
import { Frame } from 'src/ui/ui-kit/Frame';
```

File: src/ui/pages/Security/AutoLockTimer.tsx [hunk 2]
```diff
@@ -41,49 +46,74 @@ export const AUTO_LOCK_TIMER_OPTIONS_TITLES = Object.fromEntries(
...
-            isInteractive: true,
-            pad: false,
-            component: (
-              <SurfaceItemLabel>
-                <UIText
-                  kind="body/regular"
-                  color={
-                    value === preference.value ? 'var(--primary)' : undefined
-                  }
+    <Background backgroundKind="white">
```
Comment (everdimension) [inline src/ui/pages/Security/AutoLockTimer.tsx:56]: I consider `<Background />` deprecated, use `useBackgroundKind(...)` instead. It does the same things but doesn't affect indentation

File: src/ui/pages/Security/AutoLockTimer.tsx [hunk 3]
```diff
@@ -41,49 +46,74 @@ export const AUTO_LOCK_TIMER_OPTIONS_TITLES = Object.fromEntries(
...
+                    <CheckIcon
+                      style={{ color: 'var(--primary)', width: 24, height: 24 }}
+                    />
+                  ) : isNewValue && preference.value === selectedValue ? (
+                    <CheckIcon
+                      style={{
+                        color: 'var(--neutral-800)',
+                        width: 24,
+                        height: 24,
+                      }}
```
Comment (everdimension) [inline src/ui/pages/Security/AutoLockTimer.tsx:83]: can these two ternaries be simplified?

```
{preference.value === selectedValue ?
  <CheckIcon
    style={{ color: isNewValue ?  'var(--neutral-800)' : 'var(--primary)', width: 24, height: 24 }}
  />
: null}
```

File: src/ui/pages/Security/AutoLockTimer.tsx [hunk 4]
```diff
@@ -41,49 +46,74 @@ export const AUTO_LOCK_TIMER_OPTIONS_TITLES = Object.fromEntries(
 export function AutoLockTimer() {
   const { globalPreferences, setGlobalPreferences } = useGlobalPreferences();
   const value = globalPreferences?.autoLockTimeout;
+  const [selectedValue, setSelectedValue] = useState<
+    GlobalPreferences['autoLockTimeout'] | null
+  >(null);
+
+  const isNewValue = Boolean(selectedValue && selectedValue !== value);
```
Comment (everdimension) [inline src/ui/pages/Security/AutoLockTimer.tsx:53]: why use `null` as initialValue?
Isn't this check enough?
```
  const isNewValue = selectedValue !== value;
```

File: src/ui/ui-kit/BottomPanel/BottomPanel.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+      from: {
+        y: 150,
+      },
+      to: {
+        y: show ? 0 : 150,
+      },
+    });
+
+    return (
+      <div className={styles.fixedBottomSheet} style={containerStyle}>
```
Comment (everdimension) [inline src/ui/ui-kit/BottomPanel/BottomPanel.tsx:74]: I couldn't understand why this can't be sticky, too. You would avoid weird position calculations and reliance on `--body-width`

File: src/ui/ui-kit/BottomPanel/styles.module.css [hunk 6]
```diff
@@ -0,0 +1,28 @@
+.stickyBottomSheet {
+  position: sticky;
+  bottom: 16px;
+}
+
+.fixedBottomSheet {
+  position: fixed;
+  pointer-events: none;
```
Comment (everdimension) [inline src/ui/ui-kit/BottomPanel/styles.module.css:8]: Why is this needed? I think such things should be commented at least

File: src/ui/ui-kit/Frame/styles.module.css
Comment (everdimension) [inline src/ui/ui-kit/Frame/styles.module.css:1]: I'm not sure that these styles are used in any way

And if they are, it seems they are supposed to be used for `ListItem` elements, but this coupling is not obvious. Maybe for new ListItems these styles should belong to the ListItem?

File: src/ui/ui-kit/List/index.ts
Comment (everdimension) [inline src/ui/ui-kit/List/index.ts:1]: empty file

File: src/ui/ui-kit/List/ListItem.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+export const ListItemLink = React.forwardRef(
+  (
+    { className, style, ...props }: LinkProps,
+    ref: React.Ref<HTMLAnchorElement>
+  ) => {
+    return (
+      <UnstyledLink
+        ref={ref}
+        style={{ color: 'inherit', ...style }}
+        className={cn(styles.item, 'parent-hover', className)}
```
Comment (everdimension) [inline src/ui/ui-kit/List/ListItem.tsx:18]: How is `parent-hover` class used?

File: src/ui/ui-kit/List/ListItem.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+      style,
+      ...props
+    }: React.AnchorHTMLAttributes<HTMLAnchorElement>,
+    ref: React.Ref<HTMLAnchorElement>
+  ) => {
+    return (
+      <UnstyledAnchor
+        ref={ref}
+        style={{ color: 'inherit', ...style }}
+        className={cn(styles.item, 'parent-hover', className)}
```
Comment (everdimension) [inline src/ui/ui-kit/List/ListItem.tsx:38]: How is parent-hover class used?

File: src/ui/ui-kit/List/ListItem.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+      style,
+      ...props
+    }: React.ButtonHTMLAttributes<HTMLButtonElement>,
+    ref: React.Ref<HTMLButtonElement>
+  ) => {
+    return (
+      <UnstyledButton
+        ref={ref}
+        style={{ color: 'inherit', ...style }}
+        className={cn(styles.item, 'parent-hover', className)}
```
Comment (everdimension) [inline src/ui/ui-kit/List/ListItem.tsx:58]: How is parent-hover class used?

File: src/ui/ui-kit/BottomPanel/BottomPanel.tsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+        tension: 180,
+        friction: 18,
+      },
+      from: {
+        y: 150,
+      },
+      to: {
+        y: show ? 0 : 150,
+      },
+    });
```
Comment (everdimension) [inline src/ui/ui-kit/BottomPanel/BottomPanel.tsx:71]: These can all fit on one line :)
```suggestion
    const contentStyle = useSpring({
      config: { tension: 180, friction: 18 },
      from: { y: 150 },
      to: { y: show ? 0 : 150 },
    });
```

File: src/ui/ui-kit/BottomPanel/BottomPanel.tsx [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+            right: -16,
+            ...backdropStyle,
+          }}
+        />
+      </div>
+    );
+  }
+);
+
+export const AnimatedBottomPanel = React.forwardRef(
```
Comment (everdimension) [inline src/ui/ui-kit/BottomPanel/BottomPanel.tsx:40]: I wonder how this panel's focus should be handled when it's hidden. Let's discuss

File: src/ui/ui-kit/BottomPanel/BottomPanel.tsx [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+            style={{
+              top: 24,
+              left: 16,
+              right: 16,
+              bottom: -16,
+              zIndex: -1,
+              ...backdropStyle,
+            }}
+          />
+        ) : null}
```
Comment (everdimension) [inline src/ui/ui-kit/BottomPanel/BottomPanel.tsx:92]: Why is this outside the animated element? Is it for performance or other reasons?

File: src/ui/pages/Security/Security.tsx [hunk 13]
```diff
@@ -7,22 +7,23 @@ import { HStack } from 'src/ui/ui-kit/HStack';
...
   return (
-    <PageColumn>
-      <PageTop />
-      <SurfaceList
-        items={[
-          {
-            key: 0,
-            to: 'auto-lock-timer',
-            component: (
+    <Background backgroundKind="white">
```
Comment (everdimension) [inline src/ui/pages/Security/Security.tsx:21]: let's use `useBackgroundKind`

File: src/ui/pages/Settings/Settings.tsx [hunk 14]
```diff
@@ -40,119 +48,154 @@ function SettingsMain() {
...
+              </ListItemAnchor>
+              <ListItemAnchor
+                href="https://app.getbeamer.com/zerion/en?category=extension"
+                target="_blank"
+                rel="noopener noreferrer"
+              >
+                <AngleRightRow kind="link">
+                  <HStack gap={8} alignItems="center">
+                    <BulbIcon />
+                    <UIText kind="body/regular">What's New</UIText>
```
Comment (everdimension) [inline src/ui/pages/Settings/Settings.tsx:147]: apostrophe

File: src/ui/pages/Settings/Settings.tsx [hunk 15]
```diff
@@ -186,64 +229,69 @@ function ToggleSettingLine({
...
+            />
+          </Frame>
+        </VStack>
+        <PageBottom />
+      </PageColumn>
+    </Background>
+  );
+}
+
+function TestnetFaucets() {
```
Comment (everdimension) [inline src/ui/pages/Settings/Settings.tsx:284]: What is this name? :)

File: src/ui/pages/Settings/Settings.tsx [hunk 16]
```diff
@@ -186,64 +229,69 @@ function ToggleSettingLine({
...
+  return (
+    <Background backgroundKind="white">
+      <PageColumn>
+        <NavigationTitle title="Developer Tools" />
```
Comment (everdimension) [inline src/ui/pages/Settings/Settings.tsx:288]: I see, it's a placeholder component. Update title?

File: src/ui/ui-kit/List/ListItem.tsx
Comment (everdimension) [inline src/ui/ui-kit/List/ListItem.tsx:1]: I think `List` and `ListItem` names are too generic. If we ever have different lists (lol), it will become confusing. Let's think of a more explicit name for these particular list items.

File: src/ui/ui-kit/List/ListItem.tsx (hunk 7 above)
Comment (zerts) [reply src/ui/ui-kit/List/ListItem.tsx:18]: This allows us to style inner elements' hover without additional props

File: src/ui/ui-kit/List/ListItem.tsx
Comment (zerts) [reply src/ui/ui-kit/List/ListItem.tsx:1]: Hm, this is fair. I will try to think about better options.

File: src/ui/ui-kit/BottomPanel/BottomPanel.tsx (hunk 12 above)
Comment (zerts) [reply src/ui/ui-kit/BottomPanel/BottomPanel.tsx:92]: This is for making animation style better, so background filter style will apply differently

File: src/ui/ui-kit/List/ListItem.tsx (hunk 9 above)
Comment (zerts) [reply src/ui/ui-kit/List/ListItem.tsx:58]: This allows us to style inner elements' hover without additional props :)

File: src/ui/ui-kit/Frame/styles.module.css
Comment (zerts) [reply src/ui/ui-kit/Frame/styles.module.css:1]: As I remember, there are some global redesign for hover background color, and I need to remember why I did it this way :)

File: src/ui/ui-kit/BottomPanel/BottomPanel.tsx (hunk 5 above)
Comment (zerts) [reply src/ui/ui-kit/BottomPanel/BottomPanel.tsx:74]: As I expected, this panel should also be in the bottom regardless parent's height.

File: src/ui/ui-kit/BottomPanel/BottomPanel.tsx (hunk 12 above)
Comment (everdimension) [reply src/ui/ui-kit/BottomPanel/BottomPanel.tsx:92]: Better in what way? :)

File: src/ui/ui-kit/List/ListItem.tsx (hunk 7 above)
Comment (zerts) [reply src/ui/ui-kit/List/ListItem.tsx:18]: No, now it is not used. Just a helper for the future

File: src/ui/ui-kit/BottomPanel/BottomPanel.tsx (hunk 12 above)
Comment (zerts) [reply src/ui/ui-kit/BottomPanel/BottomPanel.tsx]: removed

File: src/ui/ui-kit/BottomPanel/BottomPanel.tsx (hunk 11 above)
Comment (zerts) [reply src/ui/ui-kit/BottomPanel/BottomPanel.tsx]: Removed

File: src/ui/pages/Security/AutoLockTimer.tsx [hunk 17]
```diff
@@ -1,16 +1,25 @@
...
 import { PageColumn } from 'src/ui/components/PageColumn';
 import { PageTop } from 'src/ui/components/PageTop';
 import { useGlobalPreferences } from 'src/ui/features/preferences/usePreferences';
+import { Button } from 'src/ui/ui-kit/Button';
+import { Frame } from 'src/ui/ui-kit/Frame';
 import { HStack } from 'src/ui/ui-kit/HStack';
-import { Radio } from 'src/ui/ui-kit/Radio';
-import { SurfaceItemLabel, SurfaceList } from 'src/ui/ui-kit/SurfaceList';
+import { FrameListItem } from 'src/ui/ui-kit/FrameList';
+import { StickyBottomPanel } from 'src/ui/ui-kit/BottomPanel/BottomPanel';
```
Comment (everdimension) [inline src/ui/pages/Security/AutoLockTimer.tsx:14]:
```suggestion
import { StickyBottomPanel } from 'src/ui/ui-kit/BottomPanel';
```

File: src/ui/pages/Security/AutoLockTimer.tsx [hunk 18]
```diff
@@ -39,51 +50,84 @@ export const AUTO_LOCK_TIMER_OPTIONS_TITLES = Object.fromEntries(
...
+  const { globalPreferences, setGlobalPreferences, mutation } =
+    useGlobalPreferences();
+  useBackgroundKind({ kind: 'white' });
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (mutation.isSuccess) {
+      navigate(-1);
+    }
+  }, [mutation.isSuccess, navigate]);
```
Comment (everdimension) [inline src/ui/pages/Security/AutoLockTimer.tsx:62]: why not use `onSuccess` callback?

File: src/ui/ui-kit/FrameList/FrameListItem.tsx [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+export const FrameListItem = React.forwardRef(
+  (
+    { className, style, ...props }: React.HTMLAttributes<HTMLDivElement>,
+    ref: React.Ref<HTMLDivElement>
+  ) => {
+    return (
+      <div
+        ref={ref}
+        style={{ color: 'inherit', ...style }}
+        className={cn(styles.item, 'parent-hover', className)}
```
Comment (everdimension) [inline src/ui/ui-kit/FrameList/FrameListItem.tsx:18]: I don't think the `.parent-hover` class should be baked into the components. When it's needed, the consumer code can apply it

File: src/ui/ui-kit/BottomPanel/styles.module.css [hunk 20]
```diff
@@ -0,0 +1,24 @@
+.stickyBottomSheet {
+  position: sticky;
+  bottom: 0;
+  padding-bottom: 16px;
+}
+
+.content {
+  pointer-events: all;
```
Comment (everdimension) [inline src/ui/ui-kit/BottomPanel/styles.module.css:8]: Why is this needed?

**Review by everdimension — CHANGES_REQUESTED** [review]
I don't why the comment shows it's outdated.

Still a valid question: https://github.com/zeriontech/zerion-wallet-extension/pull/436#discussion_r1542318960

See other comments, too, pls

File: src/ui/ui-kit/FrameList/FrameListItem.tsx [hunk 21]
```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import cn from 'classnames';
+import type { LinkProps } from 'react-router-dom';
+import { UnstyledLink } from '../UnstyledLink';
+import { UnstyledAnchor } from '../UnstyledAnchor';
+import { UnstyledButton } from '../UnstyledButton';
+import * as styles from './styles.module.css';
+
+export const FrameListItem = React.forwardRef(
```
Comment (everdimension) [inline src/ui/ui-kit/FrameList/FrameListItem.tsx:9]: I think in design we have a lot of variations of these, i.e. "hightlighed" item, item without frame, etc
Should we consider all those states for this abstraction?

File: src/ui/shared/form-data.ts [hunk 22]
```diff
@@ -1,10 +1,10 @@
 import lodashSet from 'lodash/set';
 
-export function naiveFormDataToObject(
+export function naiveFormDataToObject<T>(
   formData: FormData,
-  modifier: (key: string, value: unknown) => unknown
+  modifier: (key: keyof T | string, value: unknown) => unknown
 ) {
-  const result: Record<string, unknown> = {};
+  const result: Partial<{ [K in keyof T]: T[K] } & Record<string, string>> = {};
```
Comment (everdimension) [inline src/ui/shared/form-data.ts:7]: 👍

File: src/ui/ui-kit/Frame/styles.module.css
Comment (zerts) [reply src/ui/ui-kit/Frame/styles.module.css:1]: Ah, I just realised. This is a .root class, for frame element. I definitely need to rename it to frame :)

File: src/ui/ui-kit/BottomPanel/styles.module.css (hunk 20 above)
Comment (zerts) [reply src/ui/ui-kit/BottomPanel/styles.module.css:8]: Forgot to remove after removing animated version

File: src/ui/pages/Security/AutoLockTimer.tsx [hunk 23]
```diff
@@ -39,51 +50,84 @@ export const AUTO_LOCK_TIMER_OPTIONS_TITLES = Object.fromEntries(
...
-        />
-      </VStack>
-      <PageBottom />
+                </FrameListItem>
+              </label>
+            ))}
+          </VStack>
+        </Frame>
+        <Spacer height={16} />
+        <StickyBottomPanel containerStyle={{ marginTop: 'auto' }}>
```
[the remark this answers is not in the document]
Comment (zerts) [reply src/ui/pages/Security/AutoLockTimer.tsx:113]: Ok, I was inspired by some other part of the code. I will remove this

File: src/ui/ui-kit/FrameList/FrameListItem.tsx (hunk 19 above)
Comment (zerts) [reply src/ui/ui-kit/FrameList/FrameListItem.tsx:18]: Ok :(

File: src/ui/pages/Security/AutoLockTimer.tsx (hunk 18 above)
Comment (zerts) [reply src/ui/pages/Security/AutoLockTimer.tsx:62]: Do we have onSuccess callback in useGlobalPreferences? Or can I add a param to the hook?

File: src/ui/ui-kit/FrameList/FrameListItem.tsx (hunk 21 above)
Comment (zerts) [reply src/ui/ui-kit/FrameList/FrameListItem.tsx:9]: Not sure about this. May be next time. Or let's just postpone this pull request

**Comment by zerts** [comment]
TODO: still think about namings

File: src/ui/pages/Security/AutoLockTimer.tsx [hunk 24]
```diff
@@ -39,51 +50,85 @@ export const AUTO_LOCK_TIMER_OPTIONS_TITLES = Object.fromEntries(
...
-  const { globalPreferences, setGlobalPreferences } = useGlobalPreferences();
-  const value = globalPreferences?.autoLockTimeout;
+  const { globalPreferences, setGlobalPreferences, mutation } =
+    useGlobalPreferences();
+  useBackgroundKind({ kind: 'white' });
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (mutation.isSuccess) {
+      navigate(-1);
```
Comment (everdimension) [inline src/ui/pages/Security/AutoLockTimer.tsx:60]: prefer `goBack()` from `useGoBack` hook

File: src/ui/pages/Security/AutoLockTimer.tsx (hunk 23 above)
[the remark this answers is not in the document]
Comment (everdimension) [reply src/ui/pages/Security/AutoLockTimer.tsx:114]: can it still be removed?

File: src/ui/pages/Networks/NetworkForm/NetworkForm.tsx [hunk 25]
```diff
@@ -113,16 +112,16 @@ function hasChanges(form: HTMLFormElement) {
   return false;
 }
 
-const parsers: Parsers = {
```
[the remark this answers is not in the document]
Comment (zerts) [reply src/ui/pages/Networks/NetworkForm/NetworkForm.tsx:116]: Agree!

File: src/ui/pages/Settings/Settings.tsx [hunk 26]
```diff
@@ -9,27 +9,38 @@ import { ViewSuspense } from 'src/ui/components/ViewSuspense';
...
 import LockIcon from 'jsx:src/ui/assets/lock.svg';
 import GlobeIcon from 'jsx:src/ui/assets/globe.svg';
-import SettingsIcon from 'jsx:src/ui/assets/settings.svg';
+import QuestionIcon from 'jsx:src/ui/assets/question-hint.svg';
+import BulbIcon from 'jsx:src/ui/assets/bulb.svg';
+import PremiumIcon from 'jsx:src/ui/assets/premium.svg';
 import DarkModeLampIcon from 'jsx:src/ui/assets/dark-mode-lamp.svg';
 import NetworksIcon from 'jsx:src/ui/assets/network.svg';
 import SecurityIcon from 'jsx:src/ui/assets/security.svg';
+import ToolsIcon from 'jsx:src/ui/assets/hummer.svg';
```
Comment (everdimension) [inline src/ui/pages/Settings/Settings.tsx:24]: Lol
```suggestion
import ToolsIcon from 'jsx:src/ui/assets/hammer.svg';
```

File: src/ui/assets/hummer.svg
Comment (everdimension) [inline src/ui/assets/hummer.svg:1]: hammer.svg

**Review by everdimension — CHANGES_REQUESTED** [review]
https://github.com/zeriontech/zerion-wallet-extension/pull/436/files#r1613281098

**Comment by everdimension** [comment]
All good besides the icon name:)
